documentation/design-docs/eventcounters.md

```diff
@@ -0,0 +1,306 @@
+
+```cs
+public class SimpleEventListener : EventListener
+{        
+    private readonly EventLevel _level = EventLevel.Verbose;
+
+    public int EventCount { get; private set; } = 0;
+
+    private int _intervalSec;
+
+    public SimpleEventListener(int intervalSec)
+    {
+        _intervalSec = intervalSec;
+    }
+
+
+    protected override void OnEventSourceCreated(EventSource source)
+    {
+        if (source.Name.Equals("System.Runtime"))
+        {
+            Dictionary<string, string> refreshInterval = new Dictionary<string, string>() 
+            {
+                { "EventCounterIntervalSec", _intervalSec.ToString() }
+            };
+            EnableEvents(source, _level, (EventKeywords)(-1), refreshInterval);
+        }
+    }
+
+    private (string Name, string Value) getRelevantMetric(IDictionary<string, object> eventPayload)
+    {
+        string counterName = "";
+        string counterMean = "";
+        string counterIncrement = "";
+        bool isIncrement = false;
+
+        foreach ( KeyValuePair<string, object> payload in eventPayload )
+        {
+            string key = payload.Key;
+            string val = payload.Value.ToString();
+
+            if (key.Equals("Name"))
+            {
+                counterName = val;
+            }
+            else if (key.Equals("Mean"))
+            {
+                counterMean = val;
+            }
+            else if (key.Equals("Increment"))
+            {
+                counterIncrement = val;
+                isIncrement = true;
+            }
+        }
+
+        if (isIncrement)
+        {
+            return (counterName, counterIncrement);
+        }
+        else
+        {
+            return (counterName, counterMean);
+        }
+    }
+
+    protected override void OnEventWritten(EventWrittenEventArgs eventData)
+    {
+        for (int i = 0; i < eventData.Payload.Count; i++)
+        {
+            IDictionary<string, object> eventPayload = eventData.Payload[i] as IDictionary<string, object>;
+
+            if (eventPayload != null)
+            {
+                var counterKV = getRelevantMetric(eventPayload);
+                Console.WriteLine($"{counterKV.Name} : {counterKV.Value}");
+            }
+        }
+    }
+}
+```
+
+As shown above, you *must* make sure the `"EventCounterIntervalSec"` argument is set in the filterPayload argument when calling `EnableEvents`. Otherwise the counters will not be able to flush out values since it doesn't know at which interval it should be getting flushed out.
+
+### Consuming out-of-proc
+
+Consuming EventCounters out-of-proc is also possible. For those that are familiar with ETW (Event Tracing for Windows), you can use ETW to capture counter data as events and view them on your ETW trace viewer (PerfView, WPA, etc.). You may also use `dotnet-counters` to consume it cross-platform via EventPipe. 
```

Comment:
Link to TraceEvent, and once we have published at least a preview, DiagnosticsClient.